File: QuickTalk.Api/DTOs/UserDTOs.cs
namespace QuickTalk.Api.DTOs
{
    public class UserDTO
    {
        public int UserID { get; set; }
        public string Username { get; set; }
        public string? DisplayName { get; set; }
    }

    public class UserProfileDTO
    {
        public int UserID { get; set; }
        public string Username { get; set; }
        public string Email { get; set; }
        public string? DisplayName { get; set; }
        public string? AvatarUrl { get; set; }
    }

    public class UserUpdateDTO
    {
        public string? Username { get; set; }
        public string? Email { get; set; }
        public string? AvatarUrl { get; set; }
    }

    public class UserChatroomDTO
    {
        public int ChatroomID { get; set; }
        public string Name { get; set; }
        public string RoomType { get; set; }
        public bool IsPrivate { get; set; }
    }
}

File: QuickTalk.Web/src/dto/UserDTOs.js
export const createUserDto = (userID, username, displayName = null) => ({
    userID: userID,
    username: username,
    displayName: displayName,
});
  
export const createUserProfileDto = (userID, username, email, displayName = null, avatarUrl = null) => ({
    userID: userID,
    username: username,
    email: email,
    displayName: displayName,
    avatarUrl: avatarUrl,
});
  
export const createUserUpdateDto = (username = null, email = null, avatarUrl = null) => ({
    username: username,
    email: email,
    avatarUrl: avatarUrl,
});
  
export const createUserChatroomDto = (chatroomID, name, roomType, isPrivate) => ({
    chatroomID: chatroomID,
    name: name,
    roomType: roomType,
    isPrivate: isPrivate,
});
  